subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -188,8 +188,8 @@ public FileVisitResult visitFile(Path file, @Nullable BasicFileAttributes attrs)
         private AbstractFileSystemLocationSnapshot snapshot(Path file, BasicFileAttributes attrs, String name) {
             try {
                 return getFileSnapshot(file, name, attrs);
-            } catch (Exception e) {
-                LOGGER.debug("Could not read file path '{}'.", file);
+            } catch (UncheckedIOException e) {
```

Comment:
This is exactly why we have `IOException` and `getFileSnapshot()` should throw that. Catching an unchecked exception smells, though. If we don't want to use the checked exception, then why not just catch `Exception`, as we did before?

Reply:
* it was an unchecked exception before as well (was explicitly caught and wrapped)
* Kotlin and Groovy have expelled checked exceptions for a reason
* making it checked would force everyone else to handle this case
would you still like me to use checked exception during hashing?

Reply:
@lptr?

Reply:
This code has been changed since, but `UncheckedIOException` is still caught. I'll follow up where it happens now.

Reply:
Yes, It was an explicit decision, as explained above. If you have strong opinions about it, let's discuss it.